### PR TITLE
Feature/api v2

### PR DESCRIPTION
**Pull request by onefriendaday**
* Adds support for api v2 and sets it as default endpoint

### CODE REVIEW

File: source/index.js [hunk 1]
```diff
@@ -159,6 +160,77 @@ class Storyblok {
     return this.accessToken
   }
 
+  insertRelations(story, fields) {
+    var enrich = (jtree) => {
```
Comment (ademarCardoso) [inline source/index.js:164]: Change `var enrich` to `const enrich`

File: source/index.js [hunk 2]
```diff
@@ -159,6 +160,77 @@ class Storyblok {
...
+      if (jtree == null) {
+        return
+      }
+      if (jtree.constructor === Array) {
+        for (var item = 0; item < jtree.length; item++) {
```
Comment (ademarCardoso) [inline source/index.js:169]: Change `var item` to `let item`

File: source/index.js [hunk 3]
```diff
@@ -159,6 +160,77 @@ class Storyblok {
...
+          enrich(jtree[item])
+        }
+      } else if (jtree.constructor === Object && jtree.component && jtree._uid) {
+        for (var treeItem in jtree) {
```
Comment (ademarCardoso) [inline source/index.js:173]: Change to `let`

File: source/index.js [hunk 4]
```diff
@@ -159,6 +160,77 @@ class Storyblok {
...
+          if (fields.indexOf(jtree.component + '.' + treeItem) > -1) {
+            if (typeof jtree[treeItem] === 'string') {
+              if (this.relations[jtree[treeItem]]) {
+                jtree[treeItem] = this.relations[jtree[treeItem]]
+              }
+            } else if (jtree[treeItem].constructor === Array) {
+              var stories = []
```
Comment (ademarCardoso) [inline source/index.js:180]: Change to `const` or `let`

File: source/index.js [hunk 5]
```diff
@@ -159,6 +160,77 @@ class Storyblok {
...
+      const relSize = responseData.rel_uuids.length
+      let chunks = []
+      const chunkSize = 50
+
+      for (let i = 0; i < relSize; i += chunkSize) {
+        const end = Math.min(relSize, i + chunkSize)
+        chunks.push(responseData.rel_uuids.slice(i, end))
+      }
+
+      for (var chunkIndex = 0; chunkIndex < chunks.length; chunkIndex++) {
```
Comment (ademarCardoso) [inline source/index.js:210]: Change to `let`

File: tests/resolveRelations.test.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      ]
+    }
+    await Storyblok.resolveRelations(singleStory, {version: 'published', resolve_relations: 'news.author'})
+
+    expect(singleStory.stories[0].content.author.name).toBe('Testcontent 24')
+  })
+
+  test('resolveRelations should insert relations of multiple stories with 2 extra api calls', async () => {
+    const uuids = ['e101b4fc-3736-4f82-8c8e-788e38d5d286']
+    for (var i = 0; i < 100; i++) {
```
Comment (ademarCardoso) [inline tests/resolveRelations.test.js:68]: Change to `let`

**Review by ademarCardoso — COMMENTED** [review]
Great job, I made some comments about the variables' delacrations, but the job is awesome 🚀

File: tests/redirect.test.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+jest.setTimeout(60000)
+
+const StoryblokClient = require('../source/index')
```
Comment (emanuelgsouza) [inline tests/redirect.test.js:3]: Could you change the import logic to use EsModules?

File: tests/resolveRelations.test.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
+jest.setTimeout(60000)
+
+const StoryblokClient = require('../source/index')
```
Comment (emanuelgsouza) [inline tests/resolveRelations.test.js:3]: Could you change the import logic to use EsModules?

**Review by emanuelgsouza — CHANGES_REQUESTED** [review]
There are some errors in the TravisCI related to 404 in tests. Could you take a look? In the rest, that's a good job.

**Review by thiagosaife — COMMENTED** [review]
Ran the tests here and everything is all right. Great job!

**Review by emanuelgsouza — APPROVED** [review]

File: source/index.js [hunk 9]
```diff
@@ -197,9 +270,22 @@ class Storyblok {
...
+        }
+
         if (params.version === 'published' && url != '/cdn/spaces/me') {
           provider.set(cacheKey, response)
         }
+
+        if (response.data.cv) {
+          cacheVersions[params.token] = response.data.cv
+
+          if (params.version == 'draft' && cacheVersions[params.token] != response.data.cv) {
```
Comment (mAAdhaTTah) [inline source/index.js:284]: second half of this conditional will always be `false` cuz you're setting it to that value the line before.

File: source/index.js [hunk 10]
```diff
@@ -220,8 +306,18 @@ class Storyblok {
     return this.client[type](url, params)
   }
 
-  newVersion() {
```
Comment (rokinsky) [inline source/index.js:223]: @onefriendaday it seems to me that you haven't updated the type definitions in `types/index.d.ts`

**Comment by rokinsky** [comment]
General note: why are you resolving relationships here? As for me, such code isn't related to the client and the backend should do it instead.

**Comment by onefriendaday** [comment]
> General note: why are you resolving relationships here? As for me, such code isn't related to the client and the backend should do it instead.

This version will use the new api endpoint v2/ where relationships are included in an additional array outside the content. This will avoid that duplicated content items are included in the content and a content item is only included once. The client then insert this relationships that come from the api into the content. This way the user can use the new api v2 without the need to update their implementations.

**Comment by rokinsky** [comment]
> > General note: why are you resolving relationships here? As for me, such code isn't related to the client and the backend should do it instead.
> 
> This version will use the new api endpoint v2/ where relationships are included in an additional array outside the content. This will avoid that duplicated content items are included in the content and a content item is only included once. The client then insert this relationships that come from the api into the content. This way the user can use the new api v2 without the need to update their implementations.

Got it, just for backward compatibility. Thanks for the explanation @onefriendaday!

**Pull request merged**